service-model-generator/modules/service-model-generator-ls-extension/src/main/java/io/ballerina/servicemodelgenerator/extension/ServiceModelGeneratorService.java

```diff
@@ -209,7 +212,7 @@ public CompletableFuture<ListenerDiscoveryResponse> getListeners(ListenerDiscove
                 Package currentPackage = project.currentPackage();
                 Module module = currentPackage.module(ModuleName.from(currentPackage.packageName()));
                 ModuleId moduleId = module.moduleId();
-                SemanticModel semanticModel = PackageUtil.getCompilation(currentPackage).getSemanticModel(moduleId);
+                SemanticModel semanticModel = currentPackage.getCompilation().getSemanticModel(moduleId);
```

Comment:
This approach is not recommended, as the `getCompilation` method is not concurrent safe.